subgraph/schema.graphql

```diff
@@ -0,0 +1,112 @@
+type Config @entity {
+  id: ID!
+  conviction: ConvictionConfig
+  voting: VotingConfig
+}
+
+type ConvictionConfig @entity {
+  id: ID!
+  decay: BigInt!
+  weight: BigInt!
+  maxRatio: BigInt!
+  pctBase: BigInt!
+  stakeToken: Token
+  totalStaked: BigInt!
+  maxStakedProposals: Int!
+  minThresholdStakePercentage: BigInt!
+  requestToken: Token
+}
+
+type VotingConfig @entity {
+    id: ID!
+    token: Token!
+    supportRequiredPct: BigInt!
+    minAcceptQuorumPct: BigInt!
+    durationBlocks: BigInt!
+    bufferBlocks: BigInt!
+    executionDelayBlocks: BigInt!
+}
+
+type Proposal @entity {
+  id: ID!
+  number: BigInt!
+  creator: Bytes!
+  status: ProposalStatus!
+  type: ProposalType!
+  createdAt: BigInt!
+  
+  # Proposal data
+  name: String
+  link: String
+  stakes: [Stake!] @derviedFrom(field: "proposal")
+  beneficiary: Bytes
+  requestedAmount: BigInt
+  totalTokensStaked: BigInt
+
+  # Voting data
+  metadata: String
+  startBlock: BigInt
+  executionBlock: BigInt
+  snapshotBlock: BigInt
+  supportRequiredPct: BigInt
+  minAcceptQuorum: BigInt
+  yea: BigInt
+  nay: BigInt
+  votingPower: BigInt
+  script: Bytes
+  casts: [Cast!] @derivedFrom(field: "proposal")
+}
+
+type Cast @entity {
+  id: ID!
+  voter: Bytes!
+  supports: Boolean!
+  voterStake: BigInt!
+  proposal: Proposal!
+  createdAt: BigInt!
+}
+
+type Stake @entity {
+  id: ID!
+  type: StakeType!
+  entity: Bytes!
+  proposal: Proposal!
+  amount: BigInt!
+  createdAt: BigInt!
+}
+
+type StakeHistory @entity {
+  id: ID!
+  type: StakeType!
+  entity: Bytes!
+  proposal: Proposal!
+  tokensStaked: BigInt!
+  totalTokensStaked: BigInt!
+  conviction: BigInt!
+  time: BigInt!      # Block at which was created
+  createdAt: BigInt! # Timestamp at which was created
+}
+
+type Token @entity {
+  id:ID!
+  name: String!
+  symbol: String!
+  decimals: Int!
+}
+
+enum ProposalStatus {
+  Active
+  Cancelled
+  Executed
+}
+
+enum ProposalType {
+  Suggestion # Signaling proposal
```

Comment:
opening a question to see what naming makes more sense , not sure if suggestion makes sense, on the decision one i remember at some point luke called them "administrative votes"

Reply:
Hmm, i think using the same proposal type names as what we are going to show on the UI would make the most sense to me. https://www.figma.com/file/P06HTzSyb9hExJVpwaUPmA/Honey-Pot-2.0?node-id=62%3A1413